app/pages/lab/components/UploadButton.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react'
+
+import { createSubject, createSubjectSet } from './helpers'
+
+export default function UploadButton({
+  manifest,
+  metadata,
+  project,
+  subjects
+}) {
+  const [uploading, setUploading] = useState(false);
+  const [uploadCount, setUploadCount] = useState(0);
+
+  async function createSet() {
+    setUploading(true);
+    const subjectSet = await createSubjectSet({
+      display_name: manifest.label,
+      metadata,
+      links:
+        { project: project.id }
+    });
+
+    async function uploadSubject(subject) {
+      const { priority, ...subjectMetadata } = subject.metadata;
+      const { locations } = subject;
+      const panoptesSubject = await createSubject({
+        locations,
+        metadata: {
+          ['#priority']: priority,
+          ...metadata,
+          ...subjectMetadata
+        },
+        links:
+          { project: project.id }
+      });
+      setUploadCount(count => count + 1);
+      return panoptesSubject;
+    }
+
+    const _subjects = await Promise.all(subjects.map(uploadSubject));
```

Comment:
`Promise.allSettled` might be better here, to account for any upload errors.

Reply:
I've gone down the route of wrapping `uploadSubject`, internally, with `try… catch…` so that `Promise.all` can be guaranteed to resolve.

Reply:
Single retry is done, which uploads any snapshots in the `failed` array at the end of the initial upload.

Reply:
Recursive retries, until all uploads are complete, are implemented now.